### DISCUSSION TITLE
what is the reason for such a large memory usage as the amount of data grows?

### DISCUSSION THREAD

**Discussion by andyyumiao** [Netflix/conductor#2792] [Q&A]
We have 55745 workflows and 310,000 tasks, and now the jvm heap memory reached 5g, what is the reason for such a large memory usage as the amount of data grows?
**Details:**
Conductor server:v3.3
Storage: Mysql

**Comment by aravindanr** [comment]
Its hard to answer this decisively without knowing all details about your setup, the workflows used etc. One key point to note is that, Conductor loads the entire workflow into memory to evaluate it. Its part of the roadmap to only load the necessary parts of a workflow to evaluate it.

**Reply by andyyumiao** [reply]
> Its hard to answer this decisively without knowing all details about your setup, the workflows used etc. One key point to note is that, Conductor loads the entire workflow into memory to evaluate it. Its part of the roadmap to only load the necessary parts of a workflow to evaluate it.

Will conductor load all the completed workflows into memory？including failed, terminated, successful？

**Reply by aravindanr** [reply]
No. Completed workflows are not loaded into memory. Workflows that need to be "evaluated" (scheduling next task, check task timeout etc) are the ones that are in RUNNING state.

Also ALL workflows are never loaded in memory simultaneously. When its turn for a workflow to be evaluated, its loaded in its entirety (workflow and its tasks) into memory.
